ui/src/lib/gql/job/useSubscriptions.ts

```diff
@@ -0,0 +1,125 @@
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { useRef, useEffect, useCallback } from "react";
+
+import { OnJobStatusChangeSubscription } from "../__gen__/graphql";
+import { toJobStatus } from "../convert";
+import { useWsClient } from "../provider/GraphQLSubscriptionProvider";
+
+import { JobQueryKeys } from "./useQueries";
+
+const JOB_STATUS_SUBSCRIPTION = `
+ subscription OnJobStatusChange($jobId: ID!) {
+   jobStatus(jobId: $jobId)
+ }
+`;
+
+export const useJobStatus = (jobId: string) => {
+  const wsClient = useWsClient();
+  const queryClient = useQueryClient();
+  const isSubscribedRef = useRef(false);
+  const unSubscribedRef = useRef<(() => void) | undefined>(undefined);
+  const lastStatusRef = useRef<any>(null);
+
+  const query = useQuery({
+    queryKey: [JobQueryKeys.GetJobStatus, jobId],
```

Comment:
Don't use the queries' key. Even if its the same, I feel like we should define it here. Logs is defined like so
`
export enum LogSubscriptionKeys {
  GetLogs = "getLogs",
}
`